src/sidebar/services/annotations.js

```diff
@@ -12,6 +17,80 @@ export default function annotationsService(
 ) {
   let searchClient = null;
 
+  /**
+   * Extend new annotation objects with defaults and permissions.
+   */
+  function initialize(annotationData, now = new Date()) {
+    const defaultPrivacy = store.getDefault('annotationPrivacy');
+    const groupid = store.focusedGroupId();
+    const profile = store.profile();
+
+    const userid = profile.userid;
+    const userInfo = profile.user_info;
+
+    let permissions = defaultPermissions(userid, groupid, defaultPrivacy);
+
+    // Highlights are peculiar in that they always have private permissions
+    if (metadata.isHighlight(annotationData)) {
+      permissions = privatePermissions(userid);
+    }
+
+    return Object.assign(
+      {
+        created: now.toISOString(),
+        group: groupid,
+        permissions,
+        tags: [],
+        text: '',
+        updated: now.toISOString(),
+        user: userid,
+        user_info: userInfo,
+        $tag: window.btoa(Math.random()),
```

Comment:
Can you add a comment about why `$tag` is generated the way it is? This approach works and we've obviously been using it for a long time in the "annotator" side of the client. I would generally recommend the [`generateHexString` function](https://github.com/hypothesis/client/blob/55dbe085d8e2b122919d2914d5e958eb92b7b9c1/src/sidebar/util/random.js#L12) as a universal random ID generator in the client nowadays.

Reply:
> I would generally recommend the generateHexString function as a universal random ID generator in the client nowadays.

Thanks for the pointer! Will do!